File: python/0001 TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        number_dict = {}
        
        for index, value in enumerate(nums):
            needed = target-value
            
            if needed in number_dict:
                return [index, number_dict[needed]]
            
            number_dict[value] = index

File: javascript/0001 Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numDict = {};

    for(var i=0;i<nums.length;i++){

        const val = nums[i];

        if(numDict[target-val]!==undefined) {
            return [numDict[target-val],i]
        }

        numDict[val] = i;
    }
};
